lib_test.go

```diff
@@ -3,34 +3,49 @@ package revm_api_test
 import (
 	"math/big"
 	"testing"
+	"time"
 
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/common/hexutil"
 	revm "github.com/rethmint/revm-api"
-	testca "github.com/rethmint/revm-api/contracts/Test"
+	fibca "github.com/rethmint/revm-api/contracts/Fibonacci"
 	testutils "github.com/rethmint/revm-api/testutils"
 	types "github.com/rethmint/revm-api/types/go"
 	"github.com/stretchr/testify/require"
 )
 
 const CANCUN uint8 = 17
 
-func setupTest(t *testing.T) (revm.VM, *testutils.MockKVStore, common.Address) {
+func setupTest(t *testing.T, aot bool) (revm.VM, *testutils.MockKVStore, common.Address) {
 	kvStore := testutils.NewMockKVStore()
 	vm := revm.NewVM(CANCUN)
+
+	var compiler revm.Compiler
+	if aot {
+		compiler = revm.NewCompiler()
+		go func() {
+			compiler.Start(kvStore)
+		}()
+	}
+
 	t.Cleanup(func() {
+		time.Sleep(3 * time.Second)
 		vm.Destroy()
+		if aot {
+			compiler.Destroy()
+		}
 	})
+
 	caller := common.HexToAddress("0xe100713fc15400d1e94096a545879e7c647001e0")
 	testutils.Faucet(kvStore, caller, big.NewInt(1000000000000))
 
 	return vm, kvStore, caller
 }
 
-func Test_e2e(t *testing.T) {
-	vm, kvStore, caller := setupTest(t)
-	// Deploy Test Contract
-	txData, err := hexutil.Decode(testca.TestBin)
+func Test_e2e_aot(t *testing.T) {
```

Comment:
how about making two test case(use aot true / false)? just keep the `Test_e2e`  code and add `Test_e2e_aot` test on it

Reply:
work in progress, getting cgo error at the moment

Reply:
resolved